internal/pkg/aws/cloudwatchlogs/cloudwatchlogs.go

```diff
@@ -60,13 +62,15 @@ func New(s *session.Session) *CloudWatchLogs {
 	}
 }
 
-// logStreams returns all name of the log streams in a log group.
-func (c *CloudWatchLogs) logStreams(logGroup string, logStreams ...string) ([]string, error) {
-	resp, err := c.client.DescribeLogStreams(&cloudwatchlogs.DescribeLogStreamsInput{
+// logStreams returns all name of the log streams in a log group with optional limit and prefix filters.
+func (c *CloudWatchLogs) logStreams(logGroup string, logStreamLimit *int64, logStreamPrefixes ...string) ([]string, error) {
+	describeLogStreamsInput := &cloudwatchlogs.DescribeLogStreamsInput{
 		LogGroupName: aws.String(logGroup),
 		Descending:   aws.Bool(true),
 		OrderBy:      aws.String(cloudwatchlogs.OrderByLastEventTime),
-	})
+		Limit:        logStreamLimit,
```

Comment:
I feel like we should do pagination and do the filtering for each page and then apply this limit after that (quit querying the next page if limit is reached). Otherwise it could happen that `logStreamLimit` is 20 whereas the none of the first 20 log streams with the latest events matches the log stream prefixes, resulting no log stream returned.